packages/loot-core/src/client/modals/modalsSlice.ts

```diff
@@ -0,0 +1,628 @@
+import { createSlice, type PayloadAction } from '@reduxjs/toolkit';
+
+import { send } from '../../platform/client/fetch';
+import { type File } from '../../types/file';
+import {
+  type AccountEntity,
+  type AccountSyncSource,
+  type CategoryEntity,
+  type CategoryGroupEntity,
+  type GoCardlessToken,
+  type NewRuleEntity,
+  type RuleEntity,
+  type ScheduleEntity,
+  type TransactionEntity,
+  type UserEntity,
+  type UserAccessEntity,
+  type NewUserEntity,
+  type NoteEntity,
+} from '../../types/models';
+import { setAppState } from '../app/appSlice';
+import { signOut } from '../budgets/budgetsSlice';
+import { createAppAsyncThunk } from '../redux';
+
+const sliceName = 'modals';
+
+export type Modal =
```

Comment:
Moved the `Modal` type here and changed to a discriminated union. The `options` property will be used by the modal components so that prop types don't need to be defined twice.